src/effects/backends/builtin/pitchshifteffect.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <rubberband/RubberBandStretcher.h>
+
+#include <QMap>
+
+#include "effects/backends/effectprocessor.h"
+#include "engine/effects/engineeffect.h"
+#include "engine/effects/engineeffectparameter.h"
+#include "util/class.h"
+#include "util/defs.h"
+#include "util/sample.h"
+
+namespace RubberBand {
+class RubberBandStretcher;
+} // namespace RubberBand
+
+class PitchShiftGroupState : public EffectState {
+  public:
+    // This is the default increment from RubberBand 1.8.1.
+    static constexpr size_t kRubberBandBlockSize = 256;
+
+    PitchShiftGroupState(const mixxx::EngineParameters& engineParameters)
+            : EffectState(engineParameters) {
+        initializeBuffer();
+        audioParametersChanged(engineParameters);
+    }
+
+    virtual ~PitchShiftGroupState();
+
+    void initializeBuffer() {
+        m_retrieveBuffer[0] = SampleUtil::alloc(MAX_BUFFER_LEN);
+        m_retrieveBuffer[1] = SampleUtil::alloc(MAX_BUFFER_LEN);
+    }
+
+    void audioParametersChanged(const mixxx::EngineParameters& engineParameters) {
+        m_pRubberBand = std::make_unique<RubberBand::RubberBandStretcher>(
+                engineParameters.sampleRate(),
+                engineParameters.channelCount(),
+                RubberBand::RubberBandStretcher::OptionProcessRealTime);
+
+        m_pRubberBand->setMaxProcessSize(kRubberBandBlockSize);
```

Comment:
Shouldn't we tell RubberBand here how many samples we can ever give it?
```suggestion
        m_pRubberBand->setMaxProcessSize(engineParameters.framesPerBuffer());
```

Reply:
True, I will fix it.